examples/diff.c

```diff
@@ -290,9 +305,22 @@ static void parse_opts(struct opts *o, int argc, char *argv[])
 			o->findopts.flags |= GIT_DIFF_FIND_COPIES;
 		else if (!strcmp(a, "--find-copies-harder"))
 			o->findopts.flags |= GIT_DIFF_FIND_COPIES_FROM_UNMODIFIED;
-		else if (is_prefixed(a, "-B") || is_prefixed(a, "--break-rewrites"))
-			/* TODO: parse thresholds */
+		else if (optional_str_arg(
+				break_rewrite_arg, &args, "-B", NULL) || 
+			optional_str_arg(
+				break_rewrite_arg, &args, "--break-rewrites", NULL)) {
+
 			o->findopts.flags |= GIT_DIFF_FIND_REWRITES;
+			err = extract_break_rewrites(&m, &n, *break_rewrite_arg);
+			if (err==-1) 
```

Comment:
After giving it a second thought I found your approach to be better, because on legitimate correct testcases my function was failing. You can find my new code on the same link as above.
Please review this function here before I made a new pull request [I had made enough of them].

`
int extract_break_rewrites(int *m, int *n, const char *arg)
	{
		char m_str[10];
		char n_str[10];
		int err;
		const char *slash = strchr(arg, '/');
		if (slash != arg) {
			err = sscanf(arg, "%d", n);
			if (err != 1 || err == EOF)
				return -1;
		} else {
			*n = 0;
		}
		if (slash != NULL) {
			err = sscanf(slash + 1, "%d", m);
			if (err != 1 || err == EOF)
				return -1;
		} else {
			*m = 0;
		}
		return 1;
	}
`